### PR TITLE
Accumulo 2 API Updates

### PR DESCRIPTION
**Pull request by brianloss**
This depends on SNAPSHOT versions of the microservice API/utils and in-memory Accumulo. You will need to build the release/version2.0 branch of each of the dependent microservice projects.

Changes include:
* Replacing Connector with AccumuloClient (and closing it when done, where feasible)
* Using new, non-deprecated AccumuloInputFormat/AccumuloOutputFormat.
* Using htrace instead of accumulo trace, which is now gone.

Note that the htrace changes haven't really been tested at all.

### CODE REVIEW

File: warehouse/ingest-core/src/test/resources/datawave/ingest/mapreduce/job/IngestJob-test-type.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
   <value>ingest-test</value>
   <description>This is the type of data being ingested.</description>
 </property>
-
+<!--
```
Comment (ivakegg) [inline warehouse/ingest-core/src/test/resources/datawave/ingest/mapreduce/job/IngestJob-test-type.xml]: Lets get @keith-ratcliffe to look at this one

File: warehouse/ops-tools/config-compare/pom.xml [hunk 2]
```diff
@@ -12,6 +12,12 @@
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <!-- This is normally included via hadoop-common, but we have everything excluded in the warehouse parent. -->
+        <dependency>
+            <groupId>commons-collections</groupId>
+            <artifactId>commons-collections</artifactId>
+            <version>3.2.2</version>
```
Comment (ivakegg) [inline warehouse/ops-tools/config-compare/pom.xml]: lets move this version at least into the dependency configuration at the top level

File: warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java [hunk 3]
```diff
@@ -131,44 +129,31 @@ public static DateIndexHelper getInstance() {
...
-        return null;
-    }
-    
     /**
      * Initializes the instance with a provided update interval.
      * 
-     * @param connector
-     *            A Connector to Accumulo
+     * @param client
+     *            A client connectoin to Accumulo
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java]: connection

File: warehouse/ingest-core/src/test/resources/datawave/ingest/mapreduce/job/IngestJob-test-type.xml (hunk 1 above)
Comment (brianloss) [reply warehouse/ingest-core/src/test/resources/datawave/ingest/mapreduce/job/IngestJob-test-type.xml]: Turns out this was un-necessary. Must have been something else broken while I was in the middle of making changes. I've removed this change.

File: warehouse/ops-tools/config-compare/pom.xml (hunk 2 above)
Comment (brianloss) [reply warehouse/ops-tools/config-compare/pom.xml]: Fixed in latest commit.

File: warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java (hunk 3 above)
Comment (brianloss) [reply warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java]: Fixed in latest commit.

**Pull request merged**